api/src/pcapi/core/offers/validation.py

```diff
@@ -115,7 +115,7 @@ def check_stock_price(price: decimal.Decimal, offer: models.Offer) -> None:
 
     # Cache this part to avoid N+1 when creating many stocks on the same offer.
     cache_attribute = f"_cached_checked_custom_reimbursement_rules_{offer.id}"
-    if not getattr(flask.request, cache_attribute, False):
+    if not flask.has_request_context and getattr(flask.request, cache_attribute, False):
```

Comment:
Hmm, y aurait pas une erreur dans la condition ? On veut passer dans le bloc (et faire la vérification) si on n'a pas de requête (car on n'a alors pas de cache) ou si on en a une requête mais qu'on n'y a encore rien caché.

```suggestion
    if not flask.has_request_context() or getattr(flask.request, cache_attribute, False):
```

(Au fait, `has_request_context()` est une fonction, pas une propriété.)

Et plus bas, il faudrait aussi modifier comme suit : 

```diff
-         setattr(flask.request, cache_attribute, True)
+         if flask.has_request_context():
+            setattr(flask.request, cache_attribute, True)